packages/aws-cdk/lib/api/hotswap/monitor/logs-monitor.ts

```diff
@@ -0,0 +1,237 @@
+import * as util from 'util';
+import { CloudWatchLogs } from 'aws-sdk';
+import * as colors from 'colors/safe';
+import { error } from '../../../logging';
+import { ISDK } from '../../aws-auth';
+
+export interface WithDefaultPrinterProps {
+  /**
+   * Time that the latest hotswap operation occurred
+   *
+   * @default - local machine's current time
+   */
+  readonly hotswapTime?: Date;
+
+}
+
+interface SDKMap {
+  sdk: ISDK;
+  resources: string[];
+}
+
+export interface CloudWatchLogEvent {
+  readonly message: string;
+  readonly logGroup: string;
+  readonly logStream?: string;
+  readonly ingestionTime: Date;
+}
+
+export class CloudWatchLogEventMonitor {
+  public static withDefaultPrinter(options: WithDefaultPrinterProps = {}) {
+    const stream = process.stderr;
+    const props: PrinterProps = {
+      stream,
+    };
+
+    return new CloudWatchLogEventMonitor(new EventPrinter(props), options.hotswapTime);
+  }
+
+  private active = false;
+
+  /**
+   * Determines which events not to display
+   */
+  private startTime: number;
+
+  /**
+   * Current tick timer
+   */
+  private tickTimer?: NodeJS.Timer;
+
+  /**
+   * Set to the activity of reading the current events
+   */
+  private readPromise?: Promise<any>;
+
+  private activity: Map<string, boolean> = new Map();
+  private sdkMaps: Map<string, SDKMap> = new Map();
+
+  constructor(
+    private readonly printer: IEventPrinter,
+    hotswapTime?: Date,
+  ) {
+    this.startTime = hotswapTime?.getTime() ?? Date.now();
+  }
+
+  public start() {
+    this.active = true;
+    this.scheduleNextTick();
+    return this;
+  }
+
+  /**
+   * Adds a CloudWatch log group to read log events from
+   * Since we could be reading from multiple log groups in multiple
+   * AWS accounts, we also  need to pass an SDK object for that account
```

Comment:
I'm actually not sure this is needed. I can't think of a case where we would be running watch against multiple accounts.